core/src/main/java/io/questdb/griffin/engine/join/LtJoinNoKeyRecordCursorFactory.java

```diff
@@ -123,10 +125,19 @@ public Record getRecord() {
 
         @Override
         public boolean hasNext() {
-            if (masterCursor.hasNext()) {
+            if (isMasterHasNextPending) {
+                masterHasNext = masterCursor.hasNext();
+                isMasterHasNextPending = false;
+            }
+            if (masterHasNext) {
                 // great, we have a record no matter what
                 final long masterTimestamp = masterRecord.getTimestamp(masterTimestampIndex);
+                if (masterTimestamp < slaveTimestamp) {
```

Comment:
Could you add test(s) to exercise this branch ?

Reply:
Added a test in b8101e3e434024f2ce2f5a290c6ab238ab08ae62